python/pyarrow/table.pxi

```diff
@@ -160,6 +160,8 @@ cdef class ChunkedArray(_PandasConvertible):
             return _normalize_slice(self, key)
         elif isinstance(key, int):
             return self.getitem(key)
+        elif np.isscalar(key) and np.issubdtype(key.dtype, np.integer):
+            return self.getitem(key)
```

Comment:
Alternative is to do like `array.pxi`, and let python/cython handle the cast to integer (and let it raise error otherwise:

```suggestion
        return self.getitem(_normalize_index(key, self.chunked_array.length()))
```

(minor downside is that the error message will only speak about "integer", and not "slice", but then anything convertible to int will be accepted automatically)

Reply:
Yes, we should probably use the same solution everywhere (and possibly avoid manual handling of Numpy scalars).